src/uct/ib/base/ib_device.c

```diff
@@ -562,10 +562,31 @@ uct_ib_device_query_gid_info(uct_ib_device_t *dev, uint8_t port_num,
         }
     }
 #else
+#define V1_TYPE "IB/RoCE v1"
+#define V2_TYPE "RoCE v2"
+    char buf[16];
+
     ret = ibv_query_gid(dev->ibv_context, port_num, gid_index, &info->gid);
     if (ret == 0) {
-        info->roce_version.major = 1;
-        info->roce_version.minor = 0;
+        ret = ucs_read_file_str(buf, sizeof(buf) - 1, 1,
+                          "/sys/class/infiniband/%s/ports/%d/gid_attrs/types/%d",
+                          uct_ib_device_name(dev), port_num, gid_index);
+        if (ret > 0) {
+            if (!strncmp(buf, V1_TYPE, sizeof(V1_TYPE) - 1)) {
+                info->roce_version.major = 1;
+                info->roce_version.minor = 0;
+            } else if (!strncmp(buf, V2_TYPE, sizeof(V2_TYPE) - 1)) {
```

Comment:
is it possible to have v1.5 GID?

Reply:
What is it? Isn't it some MOFED hack?

Reply:
i'm not sure. if you can see it doesn't exist upstream kernel - can ignore